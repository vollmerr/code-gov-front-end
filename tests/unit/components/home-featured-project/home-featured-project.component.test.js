import React from 'react'
import { shallow } from 'enzyme'

import HomeFeaturedProject from 'components/home-featured-project/home-featured-project.component'

const props = {
  project: {
    image: 'test-image',
    alt: 'test-alt',
    short_name: 'short-name',
    description: 'test-desc',
    links: [
      { name: 'http-1', url: 'http://test-url' },
      { name: 'other-1', url: '/other-url' },
      { name: 'http-2', url: 'https://test-url-2' },
    ],
  },
}

let wrapper
describe('components - HomeFeaturedProject', () => {
  beforeEach(() => {
    wrapper = shallow(<HomeFeaturedProject {...props} />)
  })

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
})