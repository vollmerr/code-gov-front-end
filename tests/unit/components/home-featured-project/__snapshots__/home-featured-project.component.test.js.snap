// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components - HomeFeaturedProject should render correctly 1`] = `
<div
  className="block featured-project"
>
  <div
    className="indented"
  >
    <hr />
    <div
      className="width-third"
    >
      <h2
        className="fp-short-name"
      >
        short-name
      </h2>
      <div
        align="center"
      >
        <i
          alt="test-alt"
          className="test-image"
        />
      </div>
    </div>
    <div
      className="width-two-thirds"
    >
      <p
        className="fp-description"
      >
        test-desc
      </p>
      <span
        key="http://test-url"
      >
        <a
          href="http://test-url"
        >
          <button>
            http-1
          </button>
        </a>
      </span>
      <span
        key="/other-url"
      >
        <Connect(CustomLinkComponent)
          to="/other-url"
        >
          <button>
            other-1
          </button>
        </Connect(CustomLinkComponent)>
      </span>
      <span
        key="https://test-url-2"
      >
        <a
          href="https://test-url-2"
        >
          <button>
            http-2
          </button>
        </a>
      </span>
    </div>
  </div>
</div>
`;
